cmd/zpool/zpool_main.c

```diff
@@ -2198,45 +2247,53 @@ do_import(nvlist_t *config, const char *newname, const char *mntopts,
 	uint64_t state;
 	uint64_t version;
 
-	verify(nvlist_lookup_string(config, ZPOOL_CONFIG_POOL_NAME,
-	    &name) == 0);
+	name = fnvlist_lookup_string(config, ZPOOL_CONFIG_POOL_NAME);
+	state = fnvlist_lookup_uint64(config, ZPOOL_CONFIG_POOL_STATE);
+	version = fnvlist_lookup_uint64(config, ZPOOL_CONFIG_VERSION);
 
-	verify(nvlist_lookup_uint64(config,
-	    ZPOOL_CONFIG_POOL_STATE, &state) == 0);
-	verify(nvlist_lookup_uint64(config,
-	    ZPOOL_CONFIG_VERSION, &version) == 0);
 	if (!SPA_VERSION_IS_SUPPORTED(version)) {
 		(void) fprintf(stderr, gettext("cannot import '%s': pool "
 		    "is formatted using an unsupported ZFS version\n"), name);
 		return (1);
-	} else if (state != POOL_STATE_EXPORTED &&
+	} else if (zfs_force_import_required(config) &&
```

Comment:
Does this check need to change? If we are performing a `zpool import -f` and activity is detected, we shouldn't be allowing an import I thought?

Maybe something like the following?

```diff
@@ -2257,8 +2257,9 @@ do_import(nvlist_t *config, const char *newname, const char *mntopts,
 		(void) fprintf(stderr, gettext("cannot import '%s': pool "
 		    "is formatted using an unsupported ZFS version\n"), name);
 		return (1);
-	} else if (zfs_force_import_required(config) &&
-	    !(flags & ZFS_IMPORT_ANY_HOST)) {
+	}
+
+	if (zfs_force_import_required(config)) {
 		uint64_t hostid = 0;
 		char *hostname = NULL;
 		char *import_hostname = NULL;
@@ -2269,6 +2270,10 @@ do_import(nvlist_t *config, const char *newname, const char *mntopts,
 			import_hostname = fnvlist_lookup_string(config,
 			    ZPOOL_CONFIG_IMPORT_HOSTNAME);
 
+		/*
+		 * regardless if force importing or not, if ZPOOL_CONFIG_HOSTNAME
+		 * exists, then we cannot import this pool.
+		 */
 		if (import_hostname) {
 			if (nvlist_exists(config, ZPOOL_CONFIG_IMPORT_HOSTID))
 				hostid = fnvlist_lookup_uint64(config,
@@ -2279,7 +2284,15 @@ do_import(nvlist_t *config, const char *newname, const char *mntopts,
 			    "0x%lx).\nExport the pool on the other system "
 			    "before importing it with 'zpool import -f'.\n"),
 			    name, import_hostname, (unsigned long) hostid);
-		} else {
+			return (1);
+		}
+
+		/*
+		 * if ZPOOL_CONFIG_HOSTNAME doesn't exist and we aren't forcing
+		 * an import, then we let the user know that a force import is
+		 * an option.
+		 */
+		if (!(flags & ZFS_IMPORT_ANY_HOST)) {
 			uint64_t timestamp = fnvlist_lookup_uint64(config,
 			    ZPOOL_CONFIG_TIMESTAMP);
 
@@ -2293,9 +2306,8 @@ do_import(nvlist_t *config, const char *newname, const char *mntopts,
 			    "The pool can be imported, use 'zpool import -f' "
 			    "to import the pool.\n"), name, hostname,
 			    (unsigned long)hostid, ctime((time_t *)&timestamp));
+			return (1);
 		}
-
-		return (1);
 	}
 
 	if (zpool_import_props(g_zfs, config, newname, props, flags) != 0)
```